Classes/PSGitHub.Classes.ps1

```diff
@@ -0,0 +1,257 @@
+        $this.Bio = $object.bio
+        $this.Blog = $object.blog
+        $this.Collaborators - $object.collaborators
+        $this.Company = $object.company
+        $this.CreatedAt = $object.created_at
+        $this.DiskUsage = $object.disk_usage
+        $this.Email = $object.email
+        $this.EventsUrl = $object.events_url
+        $this.Followers = $object.followers
+        $this.Following = $object.following
+        $this.FollowersUrl = $object.followers_url
+        $this.FollowingUrl = $object.following_url
+        $this.GistsUrl = $object.gists_url
+        $this.GravatarId = $object.gravatar_id
+        $this.Hireable = $object.hireable
+        $this.HtmlUrl = $object.html_url
+        $this.Id = $object.Id
+        $this.Location = $object.location
+        $this.Login = $object.login
+        $this.Name = $object.name
+        $this.Plan = $object.plan
+        $this.PrivateGists = $object.private_gists
+        $this.PublicGists = $object.public_gists
+        $this.PublicRepos = $object.public_repos
+        $this.OrganizationsUrl = $object.organizations_url
+        $this.ReceivedEventsUrl = $object.received_events_url
+        $this.ReposUrl = $object.repos_url
+        $this.SiteAdmin = $object.site_admin
+        $this.StarredUrl = $object.starred_url
+        $this.SubscriptionsUrl = $object.subscriptions_url
+        $this.TotalOwnedRepos = $object.total_owend_repos
+        $this.TotalPrivateRepos = $object.total_private_repos
+        $this.Type = $object.type
+        $this.UpdatedAt = $object.updated_at
+        $this.Url = $object.url
+    }
+
+    # Empty constructor.
+    GitHubOwner() { }
+}
+
+Class GitHubGistFile {
+    [String[]]$Content
+    [String]$FileName
+    [String]$Language
+    [String]$RawUrl
+    [Int]$Size
+    [String]$Type
+    [Bool]$Truncated
+
+    # These properties are hard to locate, but they are the ones we care about.
+    # <GistObject>.files.PSObject.Properties.Value
+    # If a Gist is retreived by any means other then the Id, the content is stripped, hence the invoke rest method.
+    GitHubGistFile([Object]$object) {
+        $this.FileName = $object.filename
+        $this.Language = $object.language
+        $this.RawUrl = $object.raw_url
+        $this.Size = $object.size
+        $this.Type = $object.Type
+        $this.Truncated = $object.truncated
+        $this.Content = if ([String]::IsNullOrEmpty($object.content)) { $this.GetFileContent() } else { $object.content }
+    }
+
+    # Empty Constructor.
+    GitHubGistFile() { }
+
+    # Adds a method to get the content of a Gist file.
+    [String[]] GetFileContent() {
+        return Invoke-RestMethod -Method Post -Uri $this.RawUrl
+    }
+}
+
+# Object may not be only used for Gists.
+Class GitHubGistFork {
+    [GitHubUser]$User
+
+    GitHubGistForks([Object]$object) {
+        $this.User = $object.forks
+    }
+
+    # Empty Constructor.
+    GitHubGistForks() { }
+}
+
+# Object may not be only used for Gists.
```

Comment:
I'm not sure what you asking, but I assume its the comment.  Basically the History, Fork, and ChangeStatus I have a feeling will be used in repos as well as Gists, if that is the case, I think we should name them better.  I added some better documentation to express this.